volumes/miovision/sql/function/function-update-open_issues.sql

```diff
@@ -0,0 +1,130 @@
+            string_agg(DISTINCT alerts.alert, '; '::text) AS alerts
+        FROM miovision_api.anomalous_ranges AS ar
+        JOIN miovision_api.alerts
+            ON alerts.intersection_uid = ar.intersection_uid
+            AND tsrange(alerts.start_time, alerts.end_time)
+            && tsrange(ar.range_start, ar.range_end)
+        GROUP BY ar.uid
+    ),
+
+    open_issues AS (
+        SELECT
+            ar.uid,
+            ar.intersection_uid,
+            i.id AS intersection_id,
+            i.api_name AS intersection_name,
+            ar.classification_uid,
+            CASE
+                WHEN ar.classification_uid = 2 THEN 'Bicycle TMC'
+                WHEN ar.classification_uid = 10 THEN 'Bicycle Approach'
+                WHEN ar.classification_uid IS NULL THEN 'All modes'
+                ELSE c.classification
+            END,
+            ar.leg,
+            ar.range_start::date,
+            (current_timestamp AT TIME ZONE 'EST5EDT')::date - ar.range_start::date AS num_days,
+            ar.notes,
+            SUM(v.volume) AS volume,
+            alerts.alerts
+        FROM miovision_api.anomalous_ranges AS ar
+        --keep rows with null classification_uid
+        LEFT JOIN miovision_api.classifications AS c USING (classification_uid)
+        --omit null intersection_uids. These will go under discontinuities. 
+        JOIN miovision_api.intersections AS i USING (intersection_uid)
+        --find last week volume
+        LEFT JOIN miovision_api.volumes AS v
+            ON ar.intersection_uid = v.intersection_uid
+            --volume within the last 7 days and after AR started
+            AND v.datetime_bin >= ar.range_start
+            --prune the partitions
+            AND v.datetime_bin >= current_date - interval '7 days'
+            AND (
+                ar.classification_uid = v.classification_uid
+                OR ar.classification_uid IS NULL
+            )
+            AND (
+                ar.leg = v.leg
+                OR ar.leg IS NULL
+            )
+        LEFT JOIN alerts ON alerts.uid = ar.uid
+        WHERE
+            ar.problem_level <> 'valid-caveat'
+            --currently active
+            AND (
+                ar.range_end IS NULL
+                OR (
+                    ar.notes LIKE '%identified by a daily airflow process%'
+                    AND ar.range_end = (current_timestamp AT TIME ZONE 'EST5EDT')::date --today
+                )
+            )
+        GROUP BY
+            ar.uid,
+            ar.intersection_uid,
+            i.id,
+            i.api_name,
+            ar.classification_uid,
+            c.classification,
+            ar.range_start,
+            ar.notes,
+            alerts.alerts
+        ORDER BY
+            ar.intersection_uid,
+            ar.range_start,
+            ar.classification_uid
+    ),
+
+    closed AS (
+        DELETE FROM miovision_api.open_issues_review
+        WHERE uid NOT IN (SELECT uid FROM open_issues)
+    )
```

Comment:
CTE `closed` seems not to be used after declaration? Or it is collaborating with the MERGE INTO lines below to do something since it's a DELETE CTE? :meow_dizzy_puffy:

Reply:
This `MERGE INTO` is not much different than a `INSERT ... UPDATE`. Maybe I should use that instead for readability...